File: src/models/Owner.js
const mongoose = require("mongoose");

const ownerSchema = new mongoose.Schema(
  {
    shopId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Shop",
      required: true,
      index: true
    },
    name: {
      type: String,
      required: true,
      trim: true
    },
    phone: {
      type: String,
      required: true,
      unique: true,
      trim: true
    },
    email: {
      type: String,
      lowercase: true,
      trim: true,
      default: ""
    },
    passwordHash: {
      type: String,
      default: ""
    },
    authProvider: {
      type: String,
      enum: ["password", "google"],
      default: "password"
    },
    googleSub: {
      type: String,
      trim: true,
      default: ""
    },
    isActive: {
      type: Boolean,
      default: true
    },
    lastLoginAt: Date
  },
  {
    timestamps: true
  }
);

ownerSchema.index(
  { googleSub: 1 },
  {
    unique: true,
    sparse: true,
    partialFilterExpression: { googleSub: { $type: "string", $gt: "" } }
  }
);

ownerSchema.index(
  { email: 1 },
  {
    unique: true,
    sparse: true,
    partialFilterExpression: { email: { $type: "string", $gt: "" } }
  }
);

module.exports = mongoose.model("Owner", ownerSchema);

File: owner_app/lib/src/models/owner.dart
import '../core/utils/json_utils.dart';

class Owner {
  const Owner({
    required this.id,
    required this.name,
    required this.phone,
    required this.email,
    required this.shopId,
  });

  final String id;
  final String name;
  final String phone;
  final String email;
  final String shopId;

  factory Owner.fromJson(Map<String, dynamic> json) {
    return Owner(
      id: stringFrom(json['id'] ?? json['_id']),
      name: stringFrom(json['name']),
      phone: stringFrom(json['phone']),
      email: stringFrom(json['email']),
      shopId: stringFrom(json['shopId']),
    );
  }
}

